src/shared/machine-id-setup.c

```diff
@@ -63,16 +71,17 @@ static int acquire_machine_id(const char *root, sd_id128_t *ret) {
                 return 1; /* Indicate that the machine ID is reused. */
         }
 
-        /* Then, try reading the D-Bus machine id, unless it is a symlink */
+        /* Then, try reading the D-Bus machine ID, unless it is a symlink */
         fd = chase_and_open("/var/lib/dbus/machine-id", root, CHASE_PREFIX_ROOT | CHASE_NOFOLLOW, O_RDONLY|O_CLOEXEC|O_NOCTTY, NULL);
         if (fd >= 0 && id128_read_fd(fd, ID128_FORMAT_PLAIN | ID128_REFUSE_NULL, ret) >= 0) {
                 log_info("Initializing machine ID from D-Bus machine ID.");
                 return 0;
         }
 
         if (isempty(root) && running_in_chroot() <= 0) {
+                bool machine_id_from_firmware_credentials;
                 /* Let's use a system credential for the machine ID if we can */
-                if (acquire_machine_id_from_credential(ret) >= 0)
+                if (acquire_machine_id_from_credential(ret, &machine_id_from_firmware_credentials) >= 0)
```

Comment:
all i am saying here basically is that you turn this into:

```c
sd_id128_t from_credential;
r = acquire_machine_id_from_credential(&from_credential);
if (r > 0) {
        if (!sd_id128_is_null(from_credential)) { 
                /* got a valid machine id from creds */
                *ret = from_credential;
                return 0;
        }

        /* We got a credential, and it was set to "firmware", hence definitely try that */
        machine_id_from_firmware = true;
}
```

or something like that.

Reply:
Okay, thank you for the clarification.